Controllers/Backend/ViewSnapshots.php

```diff
@@ -55,4 +58,68 @@ public function listAction()
             ['success' => true, 'data' => $data, 'total' => $total]
         );
     }
+
+    /**
+     * @throws ParameterMissingException
+     * @throws Exception
+     */
+    public function diffAction()
+    {
+        $sessionFrom = $this->Request()->getParam('sessionFrom');
+        $stepFrom = $this->Request()->getParam('stepFrom');
+        $sessionTo = $this->Request()->getParam('sessionTo', $sessionFrom);
+        $stepTo = $this->Request()->getParam('stepTo');
+
+        if (empty($sessionFrom)) {
+            throw new ParameterMissingException('sessionFrom is missing');
+        }
+
+        if (empty($stepFrom)) {
+            throw new ParameterMissingException('stepFrom is missing');
+        }
+
+        if (empty($sessionTo)) {
+            throw new ParameterMissingException('sessionTo is missing');
+        }
+
+        if (empty($stepTo)) {
+            throw new ParameterMissingException('stepTo is missing');
+        }
+
+        /** @var Diff $differ */
+        $differ = $this->get('frosh_view_snapshots.services.diff');
+        $dataFrom = $this->getSnapshotStep($sessionFrom, $stepFrom);
+        $dataTo = $this->getSnapshotStep($sessionTo, $stepTo);
+
+        $this->View()->assign([
+            'success' => true,
+            'data' => [
+                'sessionID' => $differ->diffPlain($dataFrom['sessionID'], $dataTo['sessionID'])->renderDiffToHTML(),
+                'template' => $differ->diffPlain($dataFrom['template'], $dataTo['template'])->renderDiffToHTML(),
+                'step' => $differ->diffPlain($dataFrom['step'], $dataTo['step'])->renderDiffToHTML(),
+                'variables' => $differ->diffSerialized($dataFrom['variables'], $dataTo['variables'])->renderDiffToHTML(),
+                'params' => $differ->diffJson($dataFrom['params'], $dataTo['params'])->renderDiffToHTML(),
+            ],
+        ]);
+    }
+
+    /**
+     * @throws Exception
+     */
+    protected function getSnapshotStep(string $sessionFrom, int $stepFrom): array
+    {
+        $qb = $this->getModelManager()->getDBALQueryBuilder();
```

Comment:
Wouldn't it be faster to use the `dbal_connection`?

Reply:
I :clipboard: :spaghetti: from the `listAction`. I assume not, given that the model manager was already requested somewhere else. Otherwise I expect `dbal_connection` to be faster too.
If I change it here and in `listAction` this PR would have multiple purposes and this might conflict with other changes. Therefore I would just change it for the new function.

Reply:
@kleinmann That is right but it will load the model manager first. As the model manager is never loaded by this plugin this is a detour.